components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/core/ApplicationAuthenticatorManager.java

```diff
@@ -184,4 +186,23 @@ private boolean isAuthenticationActionEnabled() {
         return  Boolean.parseBoolean((String) IdentityConfigParser.getInstance()
                 .getConfiguration().get(AUTHENTICATION_ACTION_ENABLED_PROP));
     }
+
+    public IdentityProvider getSerializableIdPByResourceId(String resourceId, String tenantDomain)
+            throws FrameworkException, IdentityProviderManagementException {
+
+        /* Remove non-serializable UserDefinedAuthenticatorEndpointConfig objects from the identityProviders in the
+         authentication context.
+         The UserDefinedAuthenticatorEndpointConfig contains the endpoint URI and authentication type for the
+         corresponding action. However, this information is not utilized in the authentication flow. Instead,
+         the action ID in the authenticator property is used to resolve the corresponding action.
+         Since the FederatedAuthenticatorConfig model is used in the IdentityProvider class, when creating a deep
+         clone of the Identity Provider, convert the UserDefinedFederatedAuthenticatorConfig object to
+         a FederatedAuthenticatorConfig instance. */
+        IdentityProviderManager manager =
+                (IdentityProviderManager) FrameworkServiceDataHolder.getInstance().getIdentityProviderManager();
+        Gson gson = new Gson();
+        return gson.fromJson(
+                gson.toJson(manager.getIdPByResourceId(resourceId, tenantDomain, false)),
+                IdentityProvider.class);
+    }
```

Comment:
Better write a unit test for this method and make sure instance type is not UserDefined type